subsys/bluetooth/controller/ll_sw/nordic/lll/lll_peripheral.c

```diff
@@ -206,12 +213,53 @@ static int prepare_cb(struct lll_prepare_param *p)
 
 	radio_tmr_tifs_set(EVENT_IFS_US);
 
+#if defined(CONFIG_BT_CTRL_DF_CONN_CTE_RX)
+#if defined(CONFIG_BT_CTLR_DF_PHYEND_OFFSET_COMPENSATION_ENABLE)
+	enum radio_end_evt_delay_state end_evt_delay;
+#endif /* CONFIG_BT_CTLR_DF_PHYEND_OFFSET_COMPENSATION_ENABLE */
+
+	if (lll->phy_rx != PHY_CODED) {
+		df_rx_cfg = &lll->df_rx_cfg;
+		df_rx_params = dbuf_latest_get(&df_rx_cfg->hdr, NULL);
+
+		if (df_rx_params->is_enabled == true) {
+			lll_df_conf_cte_rx_enable(df_rx_params->slot_durations,
+						  df_rx_params->ant_sw_len, df_rx_params->ant_ids,
+						  data_chan_use, CTE_INFO_IN_S1_BYTE);
+			lll->df_rx_cfg.chan = data_chan_use;
+		} else {
+			lll_df_conf_cte_info_parsing_enable();
+		}
+#if defined(CONFIG_BT_CTLR_DF_PHYEND_OFFSET_COMPENSATION_ENABLE)
+		end_evt_delay = END_EVT_DELAY_ENABLED;
+	} else {
+		end_evt_delay = END_EVT_DELAY_DISABLED;
+#endif /* CONFIG_BT_CTLR_DF_PHYEND_OFFSET_COMPENSATION_ENABLE */
```

Comment:
I was told by @cvinayak once to do not use `IS_ENABLED` in controller's code. I've been removing it from my commits in the past.

Reply:
Can not use `IS_ENABLED` here, `if (IS_ENABLED(...)) {...}` would fail compilation here as `end_evt_delay` is conditionally declared.

Reply:
@cvinayak Thanks for checking the details here. I didn't look into that, because I thought that this is a kind of a policy to avoid IS_ENABLED in the Controllers code. Have I misunderstood something or has it changed?

Reply:
No, there is no policy to avoid IS_ENABLED. IS_ENABLED to be used where compilation is not an issue (like missing struct members or prior declarations). IS_ENABLED is not used in preprocessor conditional compilations (no `#if IS_ENABLED(...)` ) but `if (IS_ENABLED(...) {...}` in c control flow is preferred.

Reply:
Agreed. `IS_ENABLED` should always be preferred if possible without adding any overhead. 

Vinayak is also correct in that `IS_ENABLED` cannot be used here though, as `end_evt_delay` is conditionally compiled (I had misread the code).